lib/pages/main/wallet_contents/add_wallet_connect/add_wallet_connect.dart

```diff
@@ -0,0 +1,333 @@
+        }
+      });
+
+      sessionProposalSubscription = walletConnectService
+          .onSessionProposal.stream
+          .listen((SessionProposalEvent? args) async {
+        List<String> notSupportedNetworks = [];
+        if (args != null) {
+          log(args.toString());
+          wcPairingId = args.id;
+          wcPairingMetadata = args.params.proposer.metadata;
+          //Automatic parsing (with registering events and methods)
+          wcPairingMethods = args.params.generatedNamespaces != null
+              ? args.params.generatedNamespaces!.entries.first.value.methods
+              : [];
+          wcPairingEvents = args.params.generatedNamespaces != null
+              ? args.params.generatedNamespaces!.entries.first.value.events
+              : [];
+          wcPairingNamespaces = args.params.generatedNamespaces;
+
+          List<String?> requiredNetworkIDs = [];
+          args.params.requiredNamespaces.forEach((key, value) {
+            requiredNetworkIDs.addAll(value.chains?.toList() ?? []);
+          });
+          for (var network in requiredNetworkIDs) {
+            if (network != null && network != Config.walletConnectChainId) {
+              notSupportedNetworks.add(network);
+            }
+          }
+        }
+        final invalidApp = args?.verifyContext?.validation.invalid;
+        final unknown = args?.verifyContext?.validation.unknown;
+        final scamApp = args?.verifyContext?.validation.scam;
+
+        if (pairingTimer != null) pairingTimer!.cancel();
+        bool? userhasConfirmed =
+            await Navigator.of(context).push(MaterialPageRoute<bool>(
+                builder: (BuildContext context) {
+                  return Pair(
+                    pairingId: wcPairingId!,
+                    pairingEvents: wcPairingEvents,
+                    pairingMethods: wcPairingMethods,
+                    pairingNamespaces: wcPairingNamespaces,
+                    pairingMetadata: wcPairingMetadata,
+                    unsupportedNetowrks: notSupportedNetworks,
+                    domainType: scamApp == true
+                        ? DomainType.scam
+                        : invalidApp == true
+                            ? DomainType.mismatch
+                            : unknown == true
+                                ? DomainType.unknown
+                                : DomainType.valid,
+                  );
+                },
+                fullscreenDialog: true));
+
+        if (userhasConfirmed != null && userhasConfirmed) {
+          //Accepted
+          try {
+            if (mounted) {
+              Navigator.of(context).pop();
+              final l10n = l10nOf(context);
+              _globalSnackBar.show(l10n.wcConnectionApproved);
+            }
+          } catch (e) {
+            _globalSnackBar.showError(e.toString());
+            setState(() {
+              isLoading = false;
+              if (pairingTimer != null) pairingTimer!.cancel();
```

Comment:
it looks this is already cancelled in line 139

Reply:
This may be obsolete after upgrade to reown. Check comment in ahmeds answer. Will remove it for now